src/Gaffer/FileSystemPath.cpp

```diff
@@ -542,3 +551,119 @@ PathFilterPtr FileSystemPath::createStandardFilter( const std::vector<std::strin
 
 	return result;
 }
+
+// Create a path that is aware of differences in OS naming schemes.
+// Linux and MacOS can have a relative path with no root, or a path rooted at "/" :
+// Windows can have a relative path with no root, a path rooted at a drive letter followed by a colon
+// or a path rooted at a server name and share name (UNC paths).
+
+void FileSystemPath::setFromString(const std::string &string)
+{
+	Names newNames;
+	std::string sanitizedString = string;
+
+	boost::replace_all( sanitizedString, g_windowsSeparator.c_str(), g_genericSeparator.c_str() );
+
+#ifdef _MSC_VER
+	// If `string` is coming from a PathMatcher, it will always have a single leading slash.
+	// On Windows, interpret that to be the start of a UNC path by adding an addition leading slash
+	// If it is in fact a drive letter path, the extra slash won't cause problems in
+	// the conditionals below with `newNames`.
+	if(
+		sanitizedString.size() &&
+		sanitizedString[0] == '/' &&
+		!boost::starts_with( sanitizedString, g_uncPrefix.string() )
+	)
+	{
+		sanitizedString = g_genericSeparator.string() + sanitizedString;
+	}
+#endif
+
+	StringAlgo::tokenize<InternedString>(sanitizedString, '/', back_inserter(newNames));
+
+	InternedString newRoot;
+	if ( newNames.size() && boost::regex_match( newNames[0].string(), g_driveLetterPattern ) )
+	{
+		newRoot = newNames[0];
+		newNames.erase( newNames.begin() );
+	}
+	else if ( boost::starts_with( sanitizedString, g_uncGenericPrefix.string() ) )
+	{
+		// The server name and share name are both required before the path is valid.
+		if( newNames.size() > 1 )
+		{
+			newRoot = newNames[0].string() + g_genericSeparator.string() + newNames[1].string();
```

Comment:
It's a little trickier than that, it would be `this.server/shareName`. I'm doing that because if you are in a file dialog, in a UNC path and keep clicking the up arrow to go up a level, you can get into a liminal space.

If you go all the way up to `//this.server` Windows won't show you any of the shares on that server. At least not with a typical directory listing API request. So you have to type in the server and share again, or use bookmarks if you have them setup to reset your location.

Including the server _and_ share name is meant to prevent getting into that situation.

Reply:
Now that I have it matching the Boost [decomposition scheme](https://www.boost.org/doc/libs/1_68_0/libs/filesystem/doc/reference.html#path-decomposition-table) the root is `//this.server` as you expected. I'm not _entirely_ happy with that from a user experience point of view, but thems the breaks.

It does match Nuke and Houdini behavior, so there's some precedent for allowing users into that liminal space.

Reply:
Could `FileSystemPath::doChildren()` find share names automatically when the path is `//this.server`? Or is that what you are describing as being too slow to be appropriate in your later comment?

Reply:
I'll have a look at that, I assume it's possible through some Windows API call because the built-in Windows file browser can do it. That would be a nice improvement.